src/m365/planner/commands/task/task-reference-remove.ts

```diff
@@ -0,0 +1,155 @@
+        const requestOptionsTaskDetails: any = {
+          url: `${this.resource}/v1.0/planner/tasks/${args.options.taskId}/details`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none',
+            'If-Match': etag,
+            'Prefer': 'return=representation'
+          },
+          responseType: 'json',
+          data: {
+            references: {
+              [this.customURIEncoder(url)]: null
+            }
+          }
+        };
+
+        return request.patch(requestOptionsTaskDetails);
+      })
+      .then((res: any): void => {
+        logger.log(res.references);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetailsEtagAndUrl(options: Options): Promise<{etag: string, url: string}> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(options.taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+    let url: string = options.url;
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string | undefined = response ? response['@odata.etag'] : undefined;
+
+        if (!etag) {
+          return Promise.reject(`Error fetching task details`);
+        }
+
+        if (options.alias) {
+          const alias = options.alias as string;
+          const urls: string[] = [];
+
+          _.each(response.references, (ref, key) => {
+            if (ref.alias?.toLocaleLowerCase() === alias.toLocaleLowerCase()) {
+              urls.push(decodeURIComponent(key));
+            }
+          });
+
+          if (!urls.length) {
+            return Promise.reject(`The specified reference with alias ${options.alias} does not exist`);
+          }
+  
+          if (urls.length > 1) {
+            return Promise.reject(`Multiple references with alias ${options.alias} found: ${urls}`);
+          }
+
+          url = urls[0];
+        }
+
+        return Promise.resolve({etag, url});
+      });
+  }
+
+  private customURIEncoder(value: string): string {
+    return value
+      .replace(/\%/g, '%25')
+      .replace(/\./g, '%2E')
+      .replace(/:/g, '%3A')
+      .replace(/@/g, '%40')
+      .replace(/#/g, '%23');
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      { option: '-u, --url [url]' },
+      { option: '--alias [alias]' },
+      { option: '-i, --taskId <taskId>' },
+      { option: '--confirm' }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+  
+  public validate(args: CommandArgs): boolean | string {
+    if (!args.options.url && !args.options.alias) {
+      return 'Specify either url or alias';
+    }
+
+    if (args.options.url && args.options.alias) {
+      return 'Specify either url or alias but not both';
+    }
```

Comment:
for this kind of validation we should use option set.
for example check this command https://github.com/pnp/cli-microsoft365/blob/main/src/m365/spo/commands/list/list-roleinheritance-reset.ts
in order to validate either this or this we may do something like
![image](https://user-images.githubusercontent.com/58668583/168392531-44fa7738-3bab-4dd7-a5b0-ca890f71b349.png)